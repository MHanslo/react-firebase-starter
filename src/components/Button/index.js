/**
*
* Button Component
*
*/

import React, { PropTypes } from 'react';
import classNames from 'classnames';

function Button(props) {
  const buttonClass = classNames('button', {
    'is-primary': props.isPrimary,
    'is-loading': props.isLoading
  });

  return (
    <div>
      <button className={buttonClass} onClick={props.onClick}>{props.children}</button>
    </div>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  isPrimary: PropTypes.bool,
  isLoading: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.node
};

export default Button;
