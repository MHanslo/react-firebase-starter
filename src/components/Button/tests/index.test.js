import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

/**
 * Component render
 */
it('should render without errors', () => {
  const component = renderer.create(
    <Button />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should change classes based on props', () => {
  const component = renderer.create(
    <Button isLoading isPrimary />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
