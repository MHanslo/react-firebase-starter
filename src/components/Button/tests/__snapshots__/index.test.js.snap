exports[`test should change classes based on props 1`] = `
<div>
  <button
    className="button is-primary is-loading"
    onClick={undefined} />
</div>
`;

exports[`test should render without errors 1`] = `
<div>
  <button
    className="button"
    onClick={undefined} />
</div>
`;
